internal/streamingnode/server/wal/walimpls/wal.go

```diff
@@ -0,0 +1,23 @@
+package walimpls
+
+import (
+	"context"
+
+	"github.com/milvus-io/milvus/internal/proto/streamingpb"
+	"github.com/milvus-io/milvus/internal/util/streamingutil/message"
+)
+
+type WALImpls interface {
+	// Channel returns the channel assignment info of the wal.
+	// Should be read-only.
+	Channel() *streamingpb.PChannelInfo
+
+	// Append writes a record to the log.
+	Append(ctx context.Context, msg message.MutableMessage) (message.MessageID, error)
```

Comment:
Looks like `WALImpls` missing `AppendAsync` method def

Reply:
`WALImpls` is not required to implement `AppendAsync`.
`WAL` will give a adaptor to auto implement it.